math/matrix/add.go

```diff
@@ -0,0 +1,39 @@
+package matrix
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+// Add adds the elements of the current matrix (m1) to another matrix (m2) element-wise and returns a new matrix.
+// The two matrices must have the same dimensions.
+// If the matrices have different dimensions or if an error occurs during addition, an error is returned.
+func Add[T constraints.Integer](matrix1, matrix2 Matrix[T]) (Matrix[T], error) {
```

Comment:
You seem to be using both a mixture of methods and functions to do matrix operations. Should these be consistent do you think?

I say this because `Add` can just as easily be a method on matrices rather than a function in the package since you are using the `SameDimensions` method... So what I'm trying to say is that your implementation should be consistent so that users know what to expect where 😄

Reply:
I tried doing this but as I have generalised the Matrix type for `any`, I want `Add` and other mathematical operations to be performed for `[T constraints.Integer]` ? How do I implement it ?
Should I use `func (m *Matrix[int]) Add(other Matrix[int]) (Matrix[int], error)`

Reply:
It's interesting to see that you've chosen `any` for the Matrix implementation. So I ask a question, what kind of "thing" would you store in a matrix for it to be useful? Should you create a matrix of the following type? Is it going to be useful?

```go
type MyType struct {
   something string
}
```

Reply:
No..Got it.So I should Change `any` to `constraints.Integer` . Right?

Reply:
Personally I'd like to create another constraint called `Arithmetic` in the place where we have `Integer` defined. This can include the complex numbers as well as floats might be very useful in the context of matrices. But `Integer` would indeed suffice for now 😄